src/App.tsx

```diff
@@ -7,33 +9,91 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  time: string;
+  oldName: string | {},
+  newName: string | {},
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    time: '',
+    oldName: 'Clock-0',
+    newName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
+    window.setInterval(() => {
+      this.setState({ time: new Date().toLocaleTimeString() });
+    }, 1000);
+
+    this.timerID = window.setInterval(() => {
+      this.setState(prevState => (
+        { oldName: prevState.newName }
+      ));
+
+      this.setState({ newName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('mousedown', this.clockShow);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(prevProps: {}, prevState: {}) {
```

Comment:
the `prevState` variables has type of `State`
```suggestion
  componentDidUpdate(prevProps: unknown, prevState: State) {
```